### PR TITLE
Rename github-authentication to github

### PR DESCRIPTION
**Pull request by joaomoreno**
This addresses the plan item:

> Make clone from/publish to GitHub generally available 

From https://github.com/microsoft/vscode-pull-request-github/issues/1788

After the rename, we can start adding features in the extension.

---

@RMacfarlane Need your eyes on this since renaming the extension from with an auth provider comes might have unforeseen consequences. What do you think?

### CODE REVIEW

File: extensions/github/src/githubServer.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export class GitHubServer {
 		this.updateStatusBarItem(true);
 
 		const state = uuid();
-		const callbackUri = await vscode.env.asExternalUri(vscode.Uri.parse(`${vscode.env.uriScheme}://vscode.github-authentication/did-authenticate`));
+		const callbackUri = await vscode.env.asExternalUri(vscode.Uri.parse(`${vscode.env.uriScheme}://vscode.github/did-authenticate`));
```
Comment (joaomoreno) [inline extensions/github/src/githubServer.ts:83]: @RMacfarlane Do we need to update the callback URLs also in the app registrations?

**Comment by joaomoreno** [comment]
@RMacfarlane From what I understand, there seems to be no problem renaming the extension, as far as the auth API goes since it [doesn't actually know which extension calls `registerAuthenticationProvider`](https://github.com/microsoft/vscode/blob/fc1c8f656cb0f2f06a9a1e7e1ebf173b674b92be/src/vs/workbench/api/common/extHost.api.impl.ts#L191-L191).

File: extensions/github/src/githubServer.ts (hunk 1 above)
Comment (RMacfarlane) [reply extensions/github/src/githubServer.ts:83]: Yep, that's right! GitHub actually has control over this so I will reach out on getting this updated

**Comment by joaomoreno** [comment]
Decided to take another approach:

I'll create a `github` extension and put the features there. We can consider moving the authentication provider into there at some later point in time.

**Comment by RMacfarlane** [comment]
Sounds good!

**Pull request closed without merging**